uefi/src/proto/device_path/mod.rs

```diff
@@ -266,6 +266,20 @@ impl PartialEq for DevicePathNode {
     }
 }
 
+impl<'a> TryFrom<&[u8]> for &'a DevicePathNode {
+    type Error = ByteConversionError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        let dp = <&DevicePathHeader>::try_from(bytes)?;
+        if usize::from(dp.length) <= bytes.len() {
+            unsafe {
+                return Ok(DevicePathNode::from_ffi_ptr(bytes.as_ptr().cast()));
+            }
+        }
+        Err(ByteConversionError::InvalidLength)
```

Comment:
tiny nit: idiomatic rust avoids unnecessary use of `return`, so you can write this as:
```rust
        if usize::from(dp.length) <= bytes.len() {
            unsafe {
                Ok(DevicePathNode::from_ffi_ptr(bytes.as_ptr().cast()))
            }
        } else {
            Err(ByteConversionError::InvalidLength)
        }
```

Reply:
I tried this before, but ran into this:


```
error[E0308]: mismatched types
   --> uefi/src/proto/device_path/mod.rs:276:17
    |
274 | /         if usize::from(dp.length) <= bytes.len() {
275 | |             unsafe {
276 | |                 Ok(DevicePathNode::from_ffi_ptr(bytes.as_ptr().cast()))
    | |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `Result<&DevicePathNode, _>`
277 | |             }
278 | |         }
    | |_________- expected this to be `()`
    |
    = note: expected unit type `()`
                    found enum `core::result::Result<&proto::device_path::DevicePathNode, _>`
help: consider using a semicolon here
    |
276 |                 Ok(DevicePathNode::from_ffi_ptr(bytes.as_ptr().cast()));
    |                                                                        +
help: consider using a semicolon here
    |
278 |         };
    |          +
help: you might have meant to return this value
    |
276 |                 return Ok(DevicePathNode::from_ffi_ptr(bytes.as_ptr().cast()));
    |                 ++++++                                                        +

```

seems like I need to explicitly return within an unsafe block